crates/bevy_ecs/src/query/access.rs

```diff
@@ -325,6 +354,57 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 }
 
+// A struct to express something like `Or<(With<A>, With<B>)>`.
+// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<(With<(A, B)>, With<(B, C)>)>`.
+#[derive(Clone, Eq, PartialEq)]
+struct ExpandedOrWithAccesses {
+    arr: SmallVec<[FixedBitSet; 8]>,
```

Comment:
I used `SmallVec` to avoid introducing another level of indirection. Is it worth it? Does the inline capacity make sense?

Reply:
This can only be answered with benchmarks, have you run before/afters? I can try on my machine.

Reply:
Yeah, I'd be grateful for the help with benchmarking these. I haven't run any benchmarks myself.

Reply:
Unless we have benchmarks I'd prefer to just use a Vec here for clarity.